src/widgets/combobox.ts

```diff
@@ -45,13 +44,25 @@ export class ComboBox extends Widget {
     }
 
     public async selectOptionByText(text: string): Promise<void> {
-        const options = await this.getOptions();
-        const optionIndex = options.findIndex((option) => option === text);
-        return this.selectOptionByNumber(optionIndex);
+        await this.openOptionsList();
+
+        let rows: ElementArrayFinder = this.allByCSS('.ag-cell-value');
+        let numberOfItems: number = await rows.count();
+        for (let i = 0; i < numberOfItems; i++) {
+            let optionText: string = await rows.get(i).getText();
+            if (text === optionText) {
+                await this.getOptionSelector(i).click();
+                return;
+            }
+        }
     }
 
-    public async selectOptionByNumber(i: number): Promise<void> {
+    public async selectOptionByNumber(optionIndex: number): Promise<void> {
```

Comment:
Just one question, based on our criteria -at least inside the project- to name parameters that express "index or position":
Is this 0-based or 1-based?
The think is that if it is 0-based (that is what we tend to connect with the "index" concept), the method's name would be better expressed as selectOptionByIndex.
But then, of course, there is no direct retro-compatibility.
If it was 1-based, instead, it would be better not to use the "index" word, but e.g. "optionPosition".
I guess it is 0-based, and then I would propose to reword the method's name, but this of course would "break" the usages in the project (they should change the calls)

Reply:
@Franek75, the selection is done by index and so as you well point the method name would be better expressed by selectOptionByIndex(), however this will break retro-compatibility and this is something we don't now. I would take this into consideration and try to apply it once there is a breaking change within the ComboBox widget.